File: src/main/java/com/linear/intellij/actions/RefreshTasksAction.java
package com.linear.intellij.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowManager;
import com.intellij.ui.content.Content;
import com.linear.intellij.toolwindow.LinearToolWindowPanel;
import org.jetbrains.annotations.NotNull;

public class RefreshTasksAction extends AnAction {
    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        Project project = e.getProject();
        if (project == null) return;
        
        ToolWindow toolWindow = ToolWindowManager.getInstance(project)
                .getToolWindow("Linear Tasks");
        
        if (toolWindow != null) {
            Content content = toolWindow.getContentManager().getContent(0);
            if (content != null && content.getComponent() instanceof LinearToolWindowPanel) {
                LinearToolWindowPanel panel = (LinearToolWindowPanel) content.getComponent();
                panel.loadTasks();
            }
        }
    }
    
    @Override
    public void update(@NotNull AnActionEvent e) {
        Project project = e.getProject();
        e.getPresentation().setEnabledAndVisible(project != null);
    }
}

File: src/main/kotlin/com/linear/intellij/actions/RefreshTasksAction.kt
package com.linear.intellij.actions

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.wm.ToolWindowManager
import com.linear.intellij.toolwindow.LinearToolWindowPanel

class RefreshTasksAction : AnAction() {
    override fun actionPerformed(e: AnActionEvent) {
        val project = e.project ?: return
        
        val toolWindow = ToolWindowManager.getInstance(project)
            .getToolWindow("Linear Tasks")
        
        toolWindow?.let { window ->
            val content = window.contentManager.getContent(0)
            val component = content?.component
            if (component is LinearToolWindowPanel) {
                component.loadTasks()
            }
        }
    }
    
    override fun update(e: AnActionEvent) {
        val project = e.project
        e.presentation.isEnabledAndVisible = project != null
    }
}